#163787644 Implement pep8 standards

asirvex opened the pull request:
**What does this PR do**

Implement linting standards using tox, flake8 and pylint

**Description of Task to be completed?**

Makes sure that the code meets the quality standards

**How should this be manually tested?**

$ on the commandline, run `tox`

####Related PT story
[#163787644](https://www.pivotaltracker.com/story/show/163787644)

sirghiny commented on .pylintrc:
```diff
@@ -0,0 +1,561 @@
+[MASTER]
```
Customise this file to our project. It should only have the paths and errors to ignore.

sirghiny commented on authors/settings.py:
```diff
@@ -96,16 +96,16 @@
 AUTH_PASSWORD_VALIDATORS = [
     {
-        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
+        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',     # noqa E501
```
To ignore an error that appears severally in one file, you can have a comment at the top e.g `#pylint:disable=invalid-name`

sirghiny commented on .pylintrc:
```diff
@@ -0,0 +1,85 @@
+
+ignore=CVS,migrations
+
+disable=print-statement,
```
Do we really need all these disabled? Just pick the specific ones that we can afford to ignore.

PaulinhoNdegwa commented:
@asirvex The implementation of the task is great. You have a good understanding of `pylint` and `tox` tools.

asirvex committed:
chore(authors haven): lint the code
- Make tox, pylintrc files
- Implement pep8 standards on the code
- Add tox to the travis file
[Finishes #163787644]

asirvex committed:
chore(authors haven): Implement linting and styling
- Correct a long line
[Finish #163787644]

asirvex committed:
chore(authors have): implement styling and linting
- change travis distro to xenial to run python 3.7
[Finish #163787644]

asirvex committed:
Feature(authors haven): configure pylint in tox
- Add pylint commands to tox
[Finishes #163787644]

asirvex committed:
chore(authors haven): Fix pylint errors
- Added requirements to deps to run pylint
[Finish #163787644]
